ocs_ci/ocs/bucket_utils.py

```diff
@@ -833,35 +833,45 @@ def s3_delete_bucket_website(s3_obj, bucketname):
     return s3_obj.s3_client.delete_bucket_website(Bucket=bucketname)
 
 
-def s3_put_bucket_versioning(s3_obj, bucketname, status="Enabled"):
+def s3_put_bucket_versioning(s3_obj, bucketname, status="Enabled", s3_client=None):
     """
     Boto3 client based Put Bucket Versioning function
 
     Args:
         s3_obj (obj): MCG or OBC object
         bucketname (str): Name of the bucket
         status (str): 'Enabled' or 'Suspended'. Default 'Enabled'
+        s3_client : Any s3 client resource
 
     Returns:
         dict : PutBucketVersioning response
     """
-    return s3_obj.s3_client.put_bucket_versioning(
-        Bucket=bucketname, VersioningConfiguration={"Status": status}
-    )
+    if s3_client:
```

Comment:
Versioning is out of scope for namespace buckets.

Reply:
Ins't just list_object_versions supported on namespace buckets? Basically, I have enabled versioning on resource targets(aws here)  then uploaded object directly to the target and just list these objects using namespace bucket. Isn't this supported?

Reply:
ahh if you enabled versioning on the target bucket so it's ok, just notice that namespace buckets on top of Azure do not support list_object_versions.

Reply:
Yup, will add a condition to skip object versioning if azure.